src/battle_util.c

```diff
@@ -4179,6 +4179,7 @@ u8 AbilityBattleEffects(u8 caseID, u8 battler, u16 ability, u8 special, u16 move
                 break;
             }
 
+            gBattleScripting.battler = 0;
```

Comment:
Why is this needed? gBattleScripting.battler was set at the beginning of the switch's case to the value of battler.

Reply:
Technically there's a scenario where a user adds another battle script activating before weather, or an item/ability effect that triggers on weather/terrain that iterates over battlescripting.battler